#!/usr/bin/env python

PACKAGE = 'ts800_ros'

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, str_t, double_t, bool_t, int_t

gen = ParameterGenerator()

gen.add("motion_control_psi_gain", double_t, 0, "info - motion controller Psi Gain", 1.2, 0, 100)
gen.add("motion_control_cross_gain1", double_t, 0, "info - motion controller Cross distance Gain 1 ", 1.5, 0, 100)
gen.add("motion_control_cross_gain2", double_t, 0, "info - motion controller Cross distance Gain 2 (inside atan2)", 1.0, 0, 100)
gen.add("motion_control_p", double_t, 0, "info - motion control kP", 40, 0, 1000)
gen.add("motion_control_i", double_t, 0, "info - motion control kI", 0, 0, 1000)
gen.add("motion_control_d", double_t, 0, "info - motion control kD", 1200, 0, 1200)
gen.add("angle_control_p", double_t, 0, "info - angle control kP", 3, 0, 100)
gen.add("angle_control_i", double_t, 0, "info - angle control kI", 0, 0, 100)
gen.add("angle_control_d", double_t, 0, "info - angle control kD", 20, 0, 100)
gen.add("lookahead", double_t, 0, "info - lookahead distance", 0.15, 0, 10)
gen.add("max_steering_angle", double_t, 0, "max_steering_angle", 0, 0, 300)

gen.add("speed_fwd", double_t, 0, "speed_fwd", 0, 0, 300)
gen.add("speed_dummy", double_t, 0, "speed_dummy", 0, 0, 300)
gen.add("speed_t2", double_t, 0, "speed_t2", 0, 0, 300)
gen.add("speed_t3", double_t, 0, "speed_t3", 0, 0, 300)

gen.add("simplify_gridmap_opening_kernal_size", int_t, 0, "Simplify Gridmap - Opening Kernal Size", 4, 2, 10)
gen.add("simplify_gridmap_wall_thickness", int_t, 0, "Simplify Gridmap - Wall Thickness", 40, 1, 100)

gen.add("wallTrack_P_gain", double_t, 0, "wallTrack - P gain", 5, 0, 30)
gen.add("wallTrack_I_gain", double_t, 0, "wallTrack - I gain", 0, 0, 30)

gen.add("clean_line_interval", double_t, 0, "clean line - interval", 0.3, 0.1, 1.0)

gen.add("wall_follow_lpf_alpha", double_t, 0, "wall follow - lpf alpha", 0.2, 0.0, 1.0)

exit(gen.generate(PACKAGE, "ts800_app", "RobotParameter"))